### ISSUE TITLE
/lib64/libc.so.6: version `GLIBC_2.25' not found

### ISSUE THREAD

**Issue by anki-code** [xxh/xxh-shell-fish-appimage#2]
_Originally posted by @pat-s in https://github.com/xxh/xxh-shell-fish-appimage/issues/1#issuecomment-596543288_ :

> Being connected and calling `omf reload` I get
> python3: /lib64/libc.so.6: version `GLIBC_2.25' not found (required by /tmp/.mount_fishOKHG4Z/usr/lib64/libpython3.6m.so.1.0)

_Originally posted by frederickjh_:
> ./fish: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by ./fish)

> ./fish: /lib64/libc.so.6: version `GLIBC_2.17' not found (required by ./fish)

**Comment by anki-code** [comment]
@pat-s this message refer us to the process of how `fish.AppImage` was built. If it was built on modern linux version with fresh new libs this error is expected. 

Could you please tell us detailed OS version where you try to run `fish.AppImage`?

**Comment by anki-code** [comment]
Also you could try to download and test `xonsh.AppImage`:
```
wget https://github.com/niess/linuxdeploy-plugin-python/releases/download/continuous/xonsh-x86_64.AppImage
chmod +x xonsh-x86_64.AppImage
./xonsh-x86_64.AppImage
```
If it works then `fish.AppImage` building process should be improved.

**Comment by anki-code** [comment]
The best steps to fix this error:
1. Understand what are you doing and what tool raises the error
2. Understand how AppImage was build, on whitch OS with which version
3. Report to https://github.com/fish-shell/fish-shell/issues/6475 about additionally testing and changing building process

**Comment by anki-code** [comment]
Waiting for new build `fish.AppImage` from fish developers with lower `glibc` version - https://github.com/fish-shell/fish-shell/issues/6475

**Comment by anki-code** [comment]
I've built [statically linked fish binary with musl](https://github.com/xxh/fish-portable). So GLIBC dependency omitted. @frederickjh @pat-s Please try this version:
```
pip install -U xxh-xxh
xxh -V
0.6.14

xxh +RI xxh-shell-fish
xxh myhost +s xxh-shell-fish
```
@frederickjh the issue #5 should be resolved also.

**Comment by pat-s** [comment]
Thanks! Works for me when using

```
xxh myhost +s xxh-shell-fish
```

What does not work is setting `xxh-shell-fish` in `~/.xxh/.xxhc`. If I then just do `xxh myhost` I get the GLIBC error again.

**Comment by anki-code** [comment]
@pat-s Great! 

1. Please update xxh:
```
pip install -U xxh-xxh
xxh -V
0.6.15
xxh myhost
```
The default shell has changed to new version.

2. Could you please show the `.xxhc` file

**Comment by pat-s** [comment]
.xxh/.xxhc

```
xxh -V                                                                                                                                                                                                                
xxh/0.6.17
```

.xxh/.xxhc
```
hosts:
  myhost:       
    +s: xxh-shell-fish  
    +I: xxh-shell-fish  
```

```sh
xxh myhost

Packages operations done!

Unknown answer from host
Traceback (most recent call last):
  File "/usr/local/bin/xxh", line 14, in <module>
    xxh.main()
  File "/usr/local/lib/python3.7/site-packages/xxh_xxh/xxh.py", line 722, in main
    host_info = self.get_host_info()
  File "/usr/local/lib/python3.7/site-packages/xxh_xxh/xxh.py", line 317, in get_host_info
    if 'unsupported option "accept-new"' in pr['output']:
KeyError: 'output'
```

**Comment by frederickjh** [comment]
@pat-s This was solved in xxh/xxh#67. It is in version 0.6.11 and higher. You need to update xxh  to the newest version with:
```bash
pip3 install -U xxh-xxh
```

**Comment by frederickjh** [comment]
@pat-s Sorry, I just saw that you have a higher version, but the error output looks the same. Looks like there may be a regression here. I currently have version 0.6.14 and this is working on it.

**Comment by frederickjh** [comment]
> xxh/.xxhc
> 
> ```
> xxh -V                                                                                                                                                                                                                
> xxh/0.6.17
> 
> hosts:
>   myhost:       
>     +s: xxh-shell-fish  
>     +I: xxh-shell-fish  
> ```

@pat-s  I am not sure you really need the last line  in ` xxh/.xxhc` , `+I: xxh-shell-fish`.   In my experience it is enough to have the first line that specifies the shell you want to use. As long as you have the shell installed locally it will upload it the first time you xxh in to a host.

**Comment by frederickjh** [comment]
@pat-s What version of SSH are you running? What does this command return?
```bash
ssh -V
```

**Comment by frederickjh** [comment]
I updated to xxh/0.6.17 and installed the latest xxh-shell-fish locally and force updates on three diferent remote servers and cannot duplicate this.

**Comment by pat-s** [comment]
```
ssh -V                                                                                                                                                                                                                                                                                                                                                                                                                                                             (3s 312ms) │
OpenSSH_8.1p1, LibreSSL 2.7.3
```

Result is the same when leaving `+I: xxh-shell-fish` out.

**Comment by anki-code** [comment]
>   File "/usr/local/lib/python3.7/site-packages/xxh_xxh/xxh.py", line 317, in get_host_info
>     if 'unsupported option "accept-new"' in pr['output']:
> KeyError: 'output'

Looks like host info is empty. What is the host OS version? Could you please try `xxh myhost +v +if` and copy the `Host info:` and `Pexpect result:` sections.

I've made some fixes and release xxh 0.6.18, please update before testing: `pip install -U xxh-xxh`

> I am not sure you really need the last line in xxh/.xxhc , +I: xxh-shell-fish.

It's a good practice to add packages in config to avoid manually installation in the future. It described at the end of [usage section](https://github.com/xxh/xxh#usage).
